samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -1010,11 +1055,24 @@ static bool event_handler(const struct event_header *eh)
 
 	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
 		if (is_config_event(eh)) {
-			const struct config_event *event =
-				cast_config_event(eh);
+			struct config_event *event = cast_config_event(eh);
+
+			if (event->is_fetch) {
+				fetch_config(event->id, event->dyndata.data,
```

Comment:
Modifying event data here. Should we really do this?
What about other listeners and listeners order?

Reply:
handled event is now const, I modify only the newly created one